ecr-login/ecr.go

```diff
@@ -31,6 +31,9 @@ type ECRHelper struct {
 	ClientFactory api.ClientFactory
 }
 
+// ensure ECRHelper adheres to the credentials.Helper interface
+var _ credentials.Helper = (*ECRHelper)(nil)
```

Comment:
Yeah, actually from go docs - [godoc/#guarantee_satisfies_interface](https://golang.org/doc/faq#guarantee_satisfies_interface). I usually avoid except in cases like this where we want to keep clear that we fully implement interface. It does blow up in cmd/main.go but this makes it much more explicit.